### PR TITLE
Networking Working Group

### PR DESCRIPTION
**Pull request by alexandruradovici**
### Pull Request Overview

This pull request defines the charter of the networking group.

[Rendered](https://github.com/OxidosAutomotive/tock/tree/network_working_group/doc/wg/network)

### Testing Strategy

N/A

### TODO or Help Wanted

Feedback is welcome, @brghena I might have missed something from our discussion about the attributions of the group.

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [ ] Ran `make prepush`.

### CODE REVIEW

File: doc/wg/network/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+## Membership and Communication
+
+The networking working group membership is a subset of the people who have
+commit (pull request merge) permissions on the Tock repository. It is
+intended to be a smaller group that represents the major perspectives
+and issues, rather than a complete group. Group membership is decided by
+the group: the exact process is not yet determined and may organically
+evolve as the group gains momentum.
+
+The group has a weekly teleconference call. All working group members
```
Comment (lschuermann) [inline doc/wg/network/README.md:32]:
```suggestion
The group has a biweekly teleconference call. All working group members
```

Assuming we stick to what we talked about at TockWorld 6. Not strictly opposed to weekly though.

File: doc/wg/network/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+The goals of the Network Tock Working Group (NWG) are to:
+
+- design a set of interfaces (traits) for networking
+- design the user land interfaces for networking
+- define how buffer management should be implemented
+
+## Members
+
+- Alexandru Radovici, OxidOS Automotive
+- Brandon Ghena, Northwestern University
```
Comment (lschuermann) [inline doc/wg/network/README.md:18]:
```suggestion
- Branden Ghena, Northwestern University
```

**Review by lschuermann — APPROVED** [review]
Looks good to me.

File: doc/wg/network/README.md (hunk 1 above)
Comment (brghena) [reply doc/wg/network/README.md:32]: "biweekly" is actually ambiguous in English and means either twice a week or once every two weeks. Both meanings are common which really just means that the word is meaningless.

```suggestion
The group has a teleconference call once every two weeks. All working group members
```

File: doc/wg/network/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+- Leon Schuermann, Princeton University
+- Tyler Potyondy, UCSD
```
Comment (brghena) [inline doc/wg/network/README.md:20]: @tyler-potyondy I wanted to confirm that you're interested in being part of the working group (you can always leave in a couple months if it no longer makes sense to join calls).

File: doc/wg/network/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+- Working Group Charter
+- Adopted 7/28/2023
+
+## Goals
+
+The goals of the Network Tock Working Group (NWG) are to:
+
+- design a set of interfaces (traits) for networking
+- design the user land interfaces for networking
+- define how buffer management should be implemented
```
Comment (brghena) [inline doc/wg/network/README.md:13]:
```suggestion
- design a set of interfaces (traits) for networking
- design the user land interfaces for networking
- define how buffer management should be implemented
- determine other kernel infrastructure useful for networking
```

File: doc/wg/network/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+help contribute to particular topics or on-going discussions. The
+working group chair decides who beyond the working group members may
+participate in the call.
+
+The working group publishes detailed notes of its calls. These will be
+posted within a week of a call. This delay is to give participants an
+opportunity to correct any errors or better explain points that came up.
+They are intended to be a communication mechanism of the group, its
+discussions, the technical issues, and decisions, not a literal
+transcription of what is said.
```
Comment (brghena) [inline doc/wg/network/README.md:43]:
```suggestion
discussions, the technical issues, and decisions, not a literal
transcription of what is said.

## Code Purview

At this time, the network working group does not have responsibility for code in any particular directories of Tock. That may change as work progresses, with the possibility of a subdirectory within `capsules/` being created under the purview of this working group.
```

**Comment by brghena** [comment]
A couple of small proposed changes, but this looks great.

**Comment by bradjc** [comment]
Who is chair?

**Comment by brghena** [comment]
Alexandru, unless he doesn't want to, in which case I'll do it. (should be added to the writeup though)

**Comment by alexandruradovici** [comment]
> Alexandru, unless he doesn't want to, in which case I'll do it. (should be added to the writeup though)

Happy to do it with help from @brghena

File: doc/wg/network/README.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+- Alexandru Radovici (Chair), OxidOS Automotive
+- Branden Ghena, Northwestern University
+- Leon Schuermann, Princeton University
+- Tyler Potyondy, UCSD
+- Ioan-Cristian Cirstea, OxidOS Automotive
+
+## Membership and Communication
+
+The networking working group membership is a subset of the people who have
+commit (pull request merge) permissions on the Tock repository. It is
```
Comment (lschuermann) [inline doc/wg/network/README.md:27]: Is this true in practice (I think the answer is now) and do we want this to be the case? I don't see any hard requirement for this.

File: doc/wg/network/README.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+and issues, rather than a complete group. Group membership is decided by
+the group: the exact process is not yet determined and may organically
+evolve as the group gains momentum.
+
+The group has a teleconference call once every two weeks. All working group members
+participate in the call. Other people may be invited to participate to
+help contribute to particular topics or on-going discussions. The
+working group chair decides who beyond the working group members may
+participate in the call.
+
```
Comment (ppannuto) [inline doc/wg/network/README.md:38]:
```suggestion
Those looking to engage with the working group are encouraged to join the `#networking-working-group` channel on the [Tock slack](https://github.com/tock/tock/#keep-up-to-date).
```

File: doc/wg/network/README.md (hunk 6 above)
Comment (brghena) [reply doc/wg/network/README.md:27]:
```suggestion
The networking working group membership is open to Tock developers interested
in the design of network interfaces. It is
```
What about this?

File: doc/wg/network/README.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+The working group publishes detailed notes of its calls. These will be
```
Comment (bradjc) [inline doc/wg/network/README.md:39]:
```suggestion

The working group publishes detailed notes of its calls. These will be
```

Comment (brghena) [reply doc/wg/network/README.md:39]: I don't understand what the suggestion is here. The "suggested change" box has the - and + exactly identical. Unless it's that you added a newline??

**Comment by alexandruradovici** [comment]
As discussed with @brghena, I deleted Ionut Cirstea from the list and added:
- Cristian Rusu from the University of Bucharest, he is doing research in OS and using Tock
- Felix Mada from OxidOS Automotive, he is responsible for implementing the CAN driver

My contribution here will be from the Politehnica University

**Review by bradjc — DISMISSED** [review]

File: doc/wg/network/README.md (hunk 6 above)
Comment (lschuermann) [reply doc/wg/network/README.md:28]: I don‘t see this addressed in the current version.

File: doc/wg/network/README.md (hunk 7 above)
Comment (lschuermann) [reply doc/wg/network/README.md:39]: Same here, not yet addressed.

**Review by lschuermann — CHANGES_REQUESTED** [review]
If I‘m not mistaken (and GitHub doesn‘t trick me), two of the comments are not yet addressed in the current version. Especially the commit access requirement is, I think, something we don‘t want.

**Comment by alevy** [comment]
@alexandruradovici ping on the final two small changes?

**Comment by alexandruradovici** [comment]
I think this should be ready.

File: doc/wg/network/README.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
+Network Working Group (NWG)
+=================================
+
+- Working Group Charter
+- Adopted 7/28/2023
```
Comment (bradjc) [inline doc/wg/network/README.md:5]:
```suggestion
- Adopted 8/11/2023
```

**Review by bradjc — DISMISSED** [review]

**Review by bradjc — APPROVED** [review]

**Review by brghena — APPROVED** [review]

**Review by lschuermann — APPROVED** [review]

**Pull request merged**